### PR TITLE
Null vs Undefined / Command Ref update

### LINKED ISSUE #734: null vs undefined in Prisma Client
Context from our internal [Slack](https://prisma-company.slack.com/archives/CD683UX7Z/p1596469116003600?thread_ts=1596467531.001600&cid=CD683UX7Z): 
image

### LINKED ISSUE #843: Document `--version`/`-v` and `--json`
Currently not in command reference.

```
C:\Users\Jan
λ prisma --version
@prisma/cli          : 2.6.2
Current platform     : windows
Query Engine         : query-engine 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at ..\..\ProgramData\nvm\v12.6.0\node_modules\@prisma\cli\query-engine-windows.exe)
Migration Engine     : migration-engine-cli 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at ..\..\ProgramData\nvm\v12.6.0\node_modules\@prisma\cli\migration-engine-windows.exe)
Introspection Engine : introspection-core 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at ..\..\ProgramData\nvm\v12.6.0\node_modules\@prisma\cli\introspection-engine-windows.exe)
Format Binary        : prisma-fmt 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at ..\..\ProgramData\nvm\v12.6.0\node_modules\@prisma\cli\prisma-fmt-windows.exe)
Studio               : 0.272.0
```

Also: https://github.com/prisma/prisma/issues/2071

### PR DESCRIPTION
**Pull request by mhwelander**

### CODE REVIEW

File: content/03-reference/01-tools-and-interfaces/05-prisma-cli/02-command-reference.mdx [hunk 1]
```diff
@@ -45,24 +53,107 @@ Examples
...
+
+The `version` command outputs information about your current `@prisma/cli` version, platform, and engine binaries.
+
+#### Options
+
+The `version` command recognizes the following options to modify its behavior:
+
+| Option   | Required | Description                                                                                                                                         |
+| -------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `--json` | No       | Specifies the path to the desired `schema.prisma` file to be processed instead of the default path. Both absolute and relative paths are supported. |
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/05-prisma-cli/02-command-reference.mdx]: description is copy pasta I think

File: content/03-reference/01-tools-and-interfaces/05-prisma-cli/02-command-reference.mdx [hunk 2]
```diff
@@ -45,24 +53,107 @@ Examples
...
+
+</CodeWithResult>
+
+
 You can get additional help on any of the `prisma` commands by adding the `--help` flag after the command.
 
 ## Commands
 
-### init
+### <inlinecode>version</inlinecode>
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/05-prisma-cli/02-command-reference.mdx]: Would be great if this somehow could also mention `-v` as the short option of the command.
(No idea how though)

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+<TopBlock>
+
+Prisma Client differentiates between `null` and `undefined`:
+
+* `null` is a **value**
+* `undefined` means **do nothing**
+
+> **Note**: This is particularly important to account for in [a **Prisma with GraphQL context**, where `null` and `undefined` are interchangeable](#use-case-null-and-undefined-in-a-graphql-resolver).
+
+In the following example, if `emailInput` is **not** valid, the query sets `email` (a **mandatory** field) to `undefined` - which means ✔ **do not include this in the update**:
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]: I was confused that the "invalid" data is `null` (especially as this article is about null vs undefined). Not sure that is really a problem though.

Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]: Fair  - changed to 'if `emailInput` is `null`'.

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+In the following example, if `emailInput` is `null`, the query sets `email` (a **mandatory** field) to `undefined` - which means ✔ **do not include this in the update**:
+
+```ts
+const update = await prisma.user.update({
+  where: {
+    id: 1,
+  },
+  data: {
+    name: "Petunia",
+|  email: emailInput != null ? emailInput : undefined, // If null, don't include in update!
```
Comment (2color) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]: Is that `|` intentional?

Comment (mhwelander) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]:
```suggestion
|   email: emailInput != null ? emailInput : undefined, // If null, don't include in update!
```

Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]: It is - to highlight the line mentioned in the intro. Does it look odd? 🤔

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+In the following example mutation that updates a user, both `authorEmail` and `name` accept `null` - from a GraphQL perspective, this means that fields are **optional**:
+
+```ts
+type Mutation {
+  // Update author's email or name, or both - or neither!
+  updateUser(id: Int!, authorEmail: String, authorName: String): User!
+}
+```
+
+Howver, if you pass `null` values for `authorEmail` or `authorName` on to Prisma, the following will happen:
```
Comment (rahul3v) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]:
```suggestion
However, if you pass `null` values for `authorEmail` or `authorName` on to Prisma, the following will happen:
```

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+Instead, set the value of `email` and `name` to `undefined` if the input value is `null`. Doing this is the same as not updating the field at all:
+
+```ts
+updateUser: (parent, args, ctx: Context) => {
+  return ctx.prisma.user.update({
+    where: { id: Number(args.id) },
+    data: { 
+|     email: args.authorEmail != null ? args.authorEmail : undefined, // If null, do nothing
+|     name: args.name != null ? args.name : undefined // If null, do nothing
```
Comment (rahul3v) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]:
```suggestion
|     name: args.authorName != null ? args.authorName : undefined // If null, do nothing
```

**Review by rahul3v — CHANGES_REQUESTED** [review]
Bit typo error and a logical error

File: content/03-reference/01-tools-and-interfaces/05-prisma-cli/02-command-reference.mdx [hunk 7]
```diff
@@ -113,7 +236,7 @@ A file to define environment variables for your project:
 DATABASE_URL="postgresql://johndoe:johndoe@localhost:5432/mydb?schema=public"
 ```
 
-### generate
+### <inlinecode>generate</inlinecode>
 
 The `generate` command generates assets like Prisma Client based on the [`generator`](https://github.com/prisma/prisma/blob/master/docs/prisma-schema-file.md#generators-optional) and [`model`](https://github.com/prisma/prisma/blob/master/docs/data-modeling.md) blocks defined in your `prisma/schema.prisma` file.
```
Comment (rahul3v) [inline content/03-reference/01-tools-and-interfaces/05-prisma-cli/02-command-reference.mdx]:
```suggestion
The `generate` command generates assets like Prisma Client based on the [`generator`](../prisma-schema/generators) and [`data model`](../prisma-schema/data-model) blocks defined in your `prisma/schema.prisma` file.
```

**Review by rahul3v — COMMENTED** [review]
* Replace all prisma/prisma/docs links

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx (hunk 4 above)
Comment (2color) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]: No it makes sense. I just wasn't aware of this line highlighting feature :)

Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/057-null-and-undefined.mdx]: 🙏

**Pull request merged**